File: 2086-minimum-number-of-buckets-required-to-collect-rainwater-from-houses/2086-minimum-number-of-buckets-required-to-collect-rainwater-from-houses.cpp
class Solution {
public:
    int minimumBuckets(string street) {
        int n = street.size();
        int res = 0;
        for(int i = 0; i < n; i++){
            if(street[i] == 'H'){
                if(i-1>= 0 && street[i-1] == 'B'){
                    continue;
                }
                if(i+1 < n && street[i+1] == '.'){
                    res++;
                    street[i+1] = 'B';
                    continue;
                }
                if(i-1 >= 0 && street[i-1] == '.'){
                    res++;
                    street[i-1] = 'B';
                    continue;
                }
                return -1;
            }
        }
        return res;
        
    }
};

File: 2086-minimum-number-of-buckets-required-to-collect-rainwater-from-houses/2086-minimum-number-of-buckets-required-to-collect-rainwater-from-houses.py
class Solution(object):
    def minimumBuckets(self, S):
        if len(S) == 1 and S[0] == '.':
            return 0
        if len(S) == 1:
            return -1
        total = 0
        ind_last = len(S)
        for index in range(len(S)):
            prev = False
            next = False
            if S[index] == "H":    
                if index != 0 and S[index-1] == '.':
                    prev = True
                if index != len(S) - 1 and S[index+1] == '.':
                    next = True
                if not (prev or next):
                    return -1
                if next and (index - 1 != ind_last or not prev):
                    total += 1
                    ind_last = index + 1
                if prev and not next and ind_last != index - 1:
                    total += 1
                    ind_last = index - 1
        return total

        